CMakeLists.txt

```diff
@@ -46,6 +49,103 @@ macro(generate_firmware NAME)
         DEPENDS ${NAME})
 endmacro()
 
+# flags
+#-----------------------------------------------------
+if (${CMAKE_SYSTEM_NAME} STREQUAL "Arm")
+    # board CORTEXM44
+    set(CPU_FLAGS
+        -mcpu=cortex-m4
+        -mthumb
+        -march=armv7e-m
+        -mfpu=fpv4-sp-d16
+        -mfloat-abi=hard
+        )
+else()
+    set(CPU_FLAGS)
+endif()
+set(OPT_FLAGS
+    -O3
+    -g
+    -fno-strict-aliasing
+    -fno-strength-reduce
+    -fomit-frame-pointer
+    -funsafe-math-optimizations
+    -fno-builtin-printf
+    -ffunction-sections
+    -fdata-sections
+    )
+set(C_FLAGS
+    -std=gnu99
+    -fno-common
+    )
+set(CXX_FLAGS
+    -fno-exceptions
+    -fno-rtti
+    -std=gnu++0x
+    -fno-threadsafe-statics
+    )
+set(WARNINGS
+    -Wall
+    -Wextra
+    -Wdouble-promotion
+    -Wshadow
+    -Wfloat-equal
+    -Wframe-larger-than=1024
+    -Wpointer-arith
+    -Wlogical-op
+    -Wmissing-declarations
+    -Wpacked
+    -Wno-unused-parameter
+    -Werror=format-security
+    -Werror=array-bounds
+    -Wfatal-errors
+    -Wformat=1
+    -Werror=unused-but-set-variable
+    -Werror=unused-variable
+    -Werror=double-promotion
+    -Werror=reorder
+    #-Wcast-qual  - generates spurious noreturn attribute warnings, try again later
```

Comment:
They might not be spurious. Some functions in the flow codebase might incorrectly have noreturn attributes